source/common/http/filter_manager.cc

```diff
@@ -599,13 +617,17 @@ void FilterManager::decodeHeaders(ActiveStreamDecoderFilter* filter, RequestHead
     if (end_stream && buffered_request_data_ && continue_data_entry == decoder_filters_.end()) {
       continue_data_entry = entry;
     }
+    terminal_filter_decoded_end_stream =
+        (end_stream && continue_data_entry == decoder_filters_.end()) &&
```

Comment:
can you comment what this check does?  it's unclear at a glance why we need the continue_data_entry here but not below.  If it's for the body added case can't we have decodeData handle it and put this check in an else block for "no body added"?

Reply:
Done